File: C++/AR_167.cpp
#include <iostream>
#include <iomanip>
#include <fstream>
#include <vector>
#include <assert.h>
#include <stdio.h>
#include <algorithm>
using namespace std;

const int LADOS_CUADRADO = 4;

int fractales(int longitud, int numeroCuadrados){
	int sumaLongitud;
	
	if (longitud == 1) sumaLongitud = LADOS_CUADRADO * numeroCuadrados;
	else {
		sumaLongitud = fractales(longitud / 2, LADOS_CUADRADO * numeroCuadrados) + numeroCuadrados * LADOS_CUADRADO * longitud;
	}
	
	return sumaLongitud;
}


bool solve() {
	int longitud;
	cin >> longitud;
	if (!std::cin)
		return false;

	if (longitud == 0) cout << "0 \n";
	else {
		int sumaLongitud = fractales(longitud, 1);
		cout << sumaLongitud << '\n';
	}
	return true;

}

int main() {
	while (solve());

	return 0;
}

File: Java/AR_167.java
import java.util.Scanner;

public class AR_167 {
    private static final int LADOS_CUADRADO=4;

    private static int fractales(int longitud, int numeroCuadrados) {
        int ret;
        
        if (longitud==1) ret=LADOS_CUADRADO*numeroCuadrados;
        else {
            ret=fractales(longitud/2, LADOS_CUADRADO*numeroCuadrados) + 
                numeroCuadrados*LADOS_CUADRADO*longitud;
        }
        
        return ret;
    }

    private static boolean solve(Scanner scanner) {
        if (!scanner.hasNextInt()) return false;
        
        int longitud=scanner.nextInt();

        if (longitud==0) System.out.println("0");
        else {
            int sumaLongitud=fractales(longitud, 1);
            System.out.println(sumaLongitud);
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);
        while (solve(scanner));

        scanner.close();
    }
}